### PR TITLE
Issue/4579 onboarding/welcome dialog

### LINKED ISSUE #4579: CPP Onboarding: Epic Phase 2
Parent issue https://github.com/woocommerce/woocommerce-android/issues/4413

- [x] Live Account in Test Mode state UI
- [x] Live Account in Test Mode state business logic
- [x] ~Increase minimum required version of WCPay to 2.9.0 to handle BEGIN_OBJECT expected (waiting for release of wcpay 2.9.0)~ We decided not to increase the required version, because it wouldn't solve the issue for users who don't use in person payments
- [ ] Show onboarding "Collect payments with a Stripe card reader" dialog on OrderDetail (in progress @malinajirka )

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #4579

### Description

This PR introduces a welcome dialog which is shown w

hen the user taps on "Collect Payment" button for the first time (once per site) - the user gets navigated to the onboarding flow. It is not shown if the user completes the onboarding in the app settings before tapping on the collect payment button.

| Light | Dark |
|--|--|
|  |  |

@woocommerce/mobile-designers I used "Continue" button instead of "Open settings" button in Figma. Since we technically don't redirect the user to the settings I found it a bit confusing, wdyt?

### Testing instructions

1. Open an order eligible for in person payments
2. Tap on "Collect Payment" button
3. Notice the welcome dialog is shown
4. Tap on continue
5. Notice the onboarding flow is shown -> if the site is onboarded -> the flow returns back to the order detail and starts scanning for readers
6. Cancel the flow
7. Tap on Collect Payment button again
8. Notice the welcome dialog is not shown again and the scanning starts right away
9. Switch to a different site, repeat steps 1-6 
10. Switch back to the original site and verify that the dialog is not shown

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ghost** [comment]
@malinajirka This is looking good once we add the dark mode illustration. Also, the dark modal background colour looks a little dark. Could you please just double check we're using #363636 (elevation overlap 16dp)
Screen Shot 2021-09-30 at 11 23 41 am

**Comment by malinajirka** [comment]
> This is looking good once we add the dark mode illustration.

Sorry, I uploaded a wrong screenshot - we are already using the dark illustration


> Also, the dark modal background colour looks a little dark. Could you please just double check we're using #363636 (elevation overlap 16dp)

We are using the dark background on all the dialogs and I don't think it's #363636 - I'll look into it in a separate PR.

**Review by ghost — APPROVED** [review]

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt [hunk 1]
```diff
@@ -52,7 +52,8 @@ object AppPrefs {
         IS_USER_ELIGIBLE,
         USER_EMAIL,
         RECEIPT_PREFIX,
-        IS_CARD_PRESENT_ELIGIBLE
+        IS_CARD_PRESENT_ELIGIBLE,
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt]: It looks like we forgot to remove that in the PR that removes feature flag =)

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt]: Fixed in 01840219a856900891caaa2362a55d95bf35a532

**Review by kidinov — APPROVED** [review]
Great job! The code is clean and works well. Left a minor comment which is not related to the PR, so please go ahead with merging if you think it's better to address it in another PR

**Comment by kidinov** [comment]
It's been fixed already. Mering then

**Pull request merged**